crates/bevy_ecs/src/world/mod.rs

```diff
@@ -907,8 +907,10 @@ impl World {
     #[inline]
     pub fn get_resource<R: Resource>(&self) -> Option<&R> {
         let component_id = self.components.get_resource_id(TypeId::of::<R>())?;
+        let p = self.get_resource_by_id(component_id)?;
         // SAFETY: unique world access
```

Comment:
Not sure how this came to be, but this function does **not**  have unique world access. This safety comment has been incorrect.

Reply:
Pulled out into #6307. Not this PR's problem though.